src/units/abilities.cpp

```diff
@@ -2048,7 +2057,15 @@ effect::effect(const unit_ability_list& list, int def, const_attack_ptr att, boo
 		effect_list_.push_back(val.second);
 	}
 
-	composite_value_ = static_cast<int>((value_set + addition) * multiplier / divisor);
+	if(max_value && min_value) {
+		composite_value_ = std::clamp(*min_value, *max_value, static_cast<int>((value_set + addition) * multiplier / divisor));
+	} else if(max_value) {
+		composite_value_ = std::min(*max_value, static_cast<int>((value_set + addition) * multiplier / divisor));
+	} else if(min_value) {
+		composite_value_ = std::max(*min_value, static_cast<int>((value_set + addition) * multiplier / divisor));
+	} else {
+		composite_value_ = static_cast<int>((value_set + addition) * multiplier / divisor);
+	}
```

Comment:
Can divisor ever be 0 here?

Reply:
fix divisor for what if equals 0 because attribute then divisor value convert to 0.001 for don't cause crash in calculation

Reply:
If it is possible to have a divisor value of 0 here, then that should be handled above when iterating over `values_div`.

Reply:
error message exist cfg."divide" iteration

Reply:
Dividing by 0.001 removed